packages/api/src/pathfinding.ts

```
@@ -0,0 +1,321 @@
+  /**
+   * Generate navigation instructions based on edge type and context
+   */
+  private generateInstruction(edge: Edge, fromNode: Node, toNode: Node): string {
+    if (edge.instructions) {
+      return edge.instructions;
+    }
+    
+    // Generate contextual instructions
+    switch (edge.type) {
+      case EdgeType.STAIRS:
+        const direction = (fromNode.floor_plan_id || 0) < (toNode.floor_plan_id || 0) ? 'up' : 'down';
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)